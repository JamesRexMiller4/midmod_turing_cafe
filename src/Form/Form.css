.input-wrapper-div {
  width: 100%;
  height: fit-content;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.form-input {
  width: 60%;
  height: 35px;
  margin: 20px;
}

.hidden {
  display: none;
}

.error {
  color: red;
}

.submit-btn {
  width: 40%;
  height: 44px;
  border: 2px solid black;
}