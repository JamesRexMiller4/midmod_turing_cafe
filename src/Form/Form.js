import React, { Component } from 'react';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {  }
  }


  submitReservation() {

  }
  render() { 
    return (
      <div className='form-wrapper-div'>
        <h1>I AM A FORM</h1>
        <input className="form-input" type='text' name="name" placeholder="Name"></input>
        <input className="form-input" type='date' name="date" placeholder="Date(mm/dd)"></input>
        <input className="form-input" type='text' name="time" placeholder="Time"></input>
        <input className="form-input" type='number' name="guests" placeholder="Number of Guests"></input>
        <button onclick="form-input" className='submit btn'>Make Reservation</button>
      </div>
      );
  }
}
 
export default Form;