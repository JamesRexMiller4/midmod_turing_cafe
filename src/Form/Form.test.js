import React from 'react';
import {shallow} from 'enzyme';
import Form from './Form.js';
 
describe('Form', () => {
let wrapper, mockCreateReservation;
 
beforeEach(() => {
  mockCreateReservation = jest.fn();
wrapper = shallow(<Form
  mockCreateReservation={mockCreateReservation}
/>)
})
 
it('should match the snapshot with all the data passed through', () => {
  expect(wrapper).toMatchSnapshot();
});

it('upon mounting it should have some state', () => {
  let defaultState = {
    name: '',
    date: '',
    time: '',
    number: null,
    className: 'hidden'
   }

  expect(wrapper.state()).toEqual(defaultState);
});

it('should update state when handleChange is called', () => {
  let mockEvent = { target: {id: 'name', value: "Jerry"} };
  
  wrapper.instance().handleChange(mockEvent);

  expect(wrapper.state('name')).toEqual('Jerry');
});


it.skip('should call submitReservation when button is clicked', () => {
  let mockEvent = { preventDefault: jest.fn() };
  wrapper.instance().submitReservation = jest.fn();

  wrapper.find('#form-btn').simulate('click');
  expect(wrapper.instance().submitReservation(mockEvent)).toHaveBeenCalled();
  // expect(wrapper.instance().submitReservation(mockEvent)).toHaveBeenCalledTimes(1)
});

});