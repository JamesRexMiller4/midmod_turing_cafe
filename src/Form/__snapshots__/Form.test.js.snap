// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all the data passed through 1`] = `
<div
  className="form-wrapper-div"
>
  <h2
    className="hidden"
  >
    Please fill out all inputs to submit reservation
  </h2>
  <div
    className="input-wrapper-div"
  >
    <input
      className="form-input"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    <input
      className="form-input"
      id="date"
      min="2019-12-11"
      name="date"
      onChange={[Function]}
      placeholder="Date(mm/dd)"
      type="date"
    />
    <input
      className="form-input"
      id="time"
      name="time"
      onChange={[Function]}
      placeholder="Time"
      type="time"
    />
    <input
      className="form-input"
      id="number"
      name="number"
      onChange={[Function]}
      placeholder="Number of Guests"
      type="number"
    />
    <button
      className="submit-btn"
      id="form-btn"
      onClick={[Function]}
    >
      Make Reservation
    </button>
  </div>
</div>
`;
